testsuite/pytests/test_compartmental_model.py

```diff
@@ -962,6 +966,145 @@ def test_add_assign(self):
+        cm = nest.Create('cm_default')
+        cm.compartments = [
+            {"parent_idx": -1, "params": sp0},
+            {"parent_idx": 0, "params": dp0},
+        ]
+        nest.SetStatus(cm, {"V_init": v_init})
+        cm.V_init = v_init
+        # cm.V_bla = v_init
+
+        mm = nest.Create('multimeter', 1, {'record_from': ["v_comp0", "v_comp1"], 'interval': 1.})
+        nest.Connect(mm, cm)
+
+        nest.Simulate(10.)
+
+        self.assertTrue(np.allclose(mm.events["v_comp0"], -60.0))
+        self.assertTrue(np.allclose(mm.events["v_comp1"], -60.0))
+
+        # test initialization different from leak
+        SP = {'C_m': 1.00, 'g_C': 0.00, 'g_L': 0.100, 'e_L': -90.0}
+        DP = {'C_m': 0.10, 'g_C': 0.10, 'g_L': 0.010, 'e_L': -80.0}
+
+        cm = nest.Create('cm_default')
+        cm.compartments = [
+            {"parent_idx": -1, "params": sp1},
+            {"parent_idx": 0, "params": dp1},
+        ]
+        cm.V_init = v_init
+
+        mm = nest.Create('multimeter', 1, {'record_from': ["v_comp0", "v_comp1"], 'interval': .1})
+        nest.Connect(mm, cm)
+
+        nest.Simulate(10.)
+
+        self.assertTrue(np.allclose(mm.events["v_comp0"][0], -60.0, atol=2e-1))
+        self.assertTrue(np.allclose(mm.events["v_comp1"][0], -60.0, atol=2e-1))
+        self.assertFalse(np.allclose(mm.events["v_comp0"][-1], -60.0, atol=2e-1))
+        self.assertFalse(np.allclose(mm.events["v_comp1"][-1], -60.0, atol=2e-1))
+
+        # test default initialization at -70.
+        cm = nest.Create('cm_default')
+        cm.compartments = [
+            {"parent_idx": -1, "params": sp2},
+            {"parent_idx": 0, "params": dp2},
+        ]
+
+        mm = nest.Create(
+            'multimeter',
+            1,
+            {
+                'record_from': ["v_comp0", "v_comp1", "m_Na_0", "h_Na_0", "n_K_0"],
+                'interval': 1.
+            }
+        )
+        nest.Connect(mm, cm)
+
+        nest.Simulate(10.)
+
+        self.assertTrue(np.allclose(mm.events["v_comp0"][0], -70.0))
+        self.assertTrue(np.allclose(mm.events["v_comp1"][0], -70.0))
+
+        # test whether activation functions are initialized correctly
+
+        def vfun(v_, th, r, q):
+            return r * (v_ - th) / (1. - np.exp(-(v_ - th) / q))
+
+        v = -70.
+        alpha_m = vfun(v, -35.013, 0.182, 9.)  # 1/ms
+        beta_m = vfun(-v, 35.013, 0.124, 9.)  # 1/ms
+        # non-standard h activation
+        m_Na_0 = alpha_m / (alpha_m + beta_m),
+        h_Na_0 = 1. / (1. + np.exp((v + 65.) / 6.2))
+
+        # activation functions
+        alpha_n = 0.02 * (v - 25.) / (1. - np.exp(-(v - 25.) / 9.))
+        beta_n = -0.002 * (v - 25.) / (1. - np.exp((v - 25.) / 9.))
+        n_K_0 = alpha_n / (alpha_n + beta_n)
+
+        self.assertTrue(np.allclose(mm.events["m_Na_0"][0], m_Na_0))
+        self.assertTrue(np.allclose(mm.events["h_Na_0"][0], h_Na_0))
+        self.assertTrue(np.allclose(mm.events["n_K_0"][0], n_K_0))
```

Comment:
Those are too many tests for different cases. Better to have this pattern for each test: *create nodes -> create network -> simulate -> test result*. Again with **pytest fixtures** it will be cleaner.

Reply:
I would propose to implement a refactorization of the compartmental model tests in a separate PR, as it is not really related to the topic of this PR specficially, and create an associated issue outlining all problems that need to be addressed.